### PR TITLE
Add conduit methods to search legalpad documents and signatures

### PR DESCRIPTION
**Pull request by Roguelazer**
With this diff, you can search for documents or signatures in conduit. Additionally, you can request all the signatures as an attachment on a document.

Some details:

 - Adds `phid` to LegalpadDocumentSignature so that it can be exposed through the standard Conduit search interface
 - Fixes typo in LegalpadDocument search for contributorPHIDs

This is a bit too large to be an extension, and really should be upstreamed.

### CODE REVIEW

File: src/applications/legalpad/engineextension/PhabricatorLegalpadSignaturesSearchEngineAttachment.php [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  public function getAttachmentForObject($object, $data, $spec) {
+    $signers = array();
+    foreach ($object->getSignatures() as $signature) {
+      $signatures[] = array(
+        'phid' => $signature->getPHID(),
+        'signerPHID' => $signature->getSignerPHID(),
+        'exemptionPHID' => $signature->getExemptionPHID(),
+        'isExemption' => $signature->getIsExemption(),
+        'signerName' => $signature->getSignerName(),
+        'signerEmail' => $signature->getSignerEmail(),
```
Comment (mwaldt) [inline src/applications/legalpad/engineextension/PhabricatorLegalpadSignaturesSearchEngineAttachment.php:27]: Can you get the date signed in here as well? I

**Review by mwaldt — COMMENTED** [review]
Seems fine overall, just the one other bit of info I thought might be helpful to include

File: src/applications/legalpad/engineextension/PhabricatorLegalpadSignaturesSearchEngineAttachment.php (hunk 1 above)
Comment (Roguelazer) [reply src/applications/legalpad/engineextension/PhabricatorLegalpadSignaturesSearchEngineAttachment.php:27]: that's a good idea!

Comment (Roguelazer) [reply src/applications/legalpad/engineextension/PhabricatorLegalpadSignaturesSearchEngineAttachment.php:27]: added

**Comment by Roguelazer** [comment]
Screenshot of my dev instance with both attachment types selected:

Screen Shot 2021-08-03 at 17 16 48

**Review by mwaldt — APPROVED** [review]
Awesome, LGTM

**Pull request merged**